### PR TITLE
Refactor gui

### PR DESCRIPTION
**Pull request by PatrikLundgren**
moved gui functionality to multiple files. Thought itd be a start towards better modularity

### CODE REVIEW

File: ViAn/GUI/drawingcomponent.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+ * Deselects all overlay tools.
+ */
+void MainWindow::deselect_overlay_tool(){
+    ui->action_rectangle->setChecked(false);
+    ui->action_circle->setChecked(false);
+    ui->action_line->setChecked(false);
+    ui->action_arrow->setChecked(false);
+    ui->action_pen->setChecked(false);
+    ui->action_text->setChecked(false);
+}
```
Comment (parso619) [inline ViAn/GUI/drawingcomponent.cpp]: This should not be in DrawingComponent. This about the toolbar.

File: ViAn/GUI/analysiscomponent.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#include "analysiscomponent.h"
+
+AnalysisComponent::AnalysisComponent()
+{
+}
```
Comment (svidem) [inline ViAn/GUI/analysiscomponent.cpp]: unessesary constructor, and uncommented

Comment (parso619) [inline ViAn/GUI/analysiscomponent.cpp]: and { on the wrong line

File: ViAn/GUI/qtreeprojectitem.cpp [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#include "qtreeprojectitem.h"
+
+QTreeProjectItem::QTreeProjectItem()
+{
+}
```
Comment (svidem) [inline ViAn/GUI/qtreeprojectitem.cpp]: Empty class

File: ViAn/GUI/videocomponent.cpp [hunk 4]
```diff
@@ -0,0 +1,335 @@
+#include "videocomponent.h"
+
+VideoComponent::VideoComponent()
+{
+}
```
Comment (svidem) [inline ViAn/GUI/videocomponent.cpp]: empty probably unused constructor

File: ViAn/GUI/videocomponent.h [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#ifndef VIDEOCOMPONENT_H
+#define VIDEOCOMPONENT_H
+#include "makeproject.h"
+class VideoComponent
+{
+public:
+    VideoComponent();
```
Comment (svidem) [inline ViAn/GUI/videocomponent.h]: Still unused class

File: ViAn/GUI/qtreeprojectitem.h [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#ifndef QTREEPROJECTITEM_H
+#define QTREEPROJECTITEM_H
+
+class QTreeProjectItem
+{
+public:
+    QTreeProjectItem();
+};
```
Comment (svidem) [inline ViAn/GUI/qtreeprojectitem.h]: Still unused class

**Pull request closed without merging**